ui.apps/src/main/content/jcr_root/apps/asset-share-commons/clientlibs/clientlib-site/js/download.js

```diff
@@ -0,0 +1,156 @@
+/*
+ * Asset Share Commons
+ *
+ * Copyright [2017]  Adobe
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/*global ContextHub: false, jQuery: false, AssetShare: false */
+
+/**
+ * Object to facilitate asynchronous download operations
+ * common operations include initiating an asynchronous download,
+ * polling for status based on downloadId,
+ * and performing the actual download once the download package is ready
+ */
+AssetShare.Download = (function ($, ns, messages, downloadStore) {
+    "use strict";
+
+    const DOWNLOAD_ID = "downloadId",
+        DWNL_STATUS_PN = "isComplete",
+        DWNL_ARTIFACTS_PN = "artifacts",
+        BODY_SELECTOR = "body.page",
+        LOADING_SELECTOR = "download-loader-text",
+        POLL_ENABLED = true,
+        POLL_ENDPOINT = "/content/dam.downloadbinaries.json",
+        POLL_TIMEOUT = 1000,
+        MAX_ATTEMPTS = 3;
+
+    /**
+     * Invoked from download Modal
+     * @param {*} form
+     */
+    function initializeDownload(form) {
+        form = $(form);
+        $.ajax({
+            type: "POST",
+            url: form.attr("action"),
+            data: form.serialize(),
+            success: function (data) {
+                if (data[DOWNLOAD_ID]) {
+                    // initialize polling
+                    if (POLL_ENABLED) {
+                        // initiate dimmer on page
+                        _showDimmer();
+                        _poll(data[DOWNLOAD_ID], 0);
+                    } else {
+                        // track downloadid in session storage
+                        _storeDownloadId(data[DOWNLOAD_ID]);
+                    }
+                }
+            },
+            error: function (e) {
+                console.error(e.responseJSON.error);
```

Comment:
Do we have to do anything special in the event of an error? (show an error, etc.)

Maybe add a stretch goal - dont think it's blocking.